test/integration/ipamperf/ipam_test.go

```diff
@@ -0,0 +1,240 @@
+		glog.Infof("Logging results to %s", resultsLogFile)
+		if err := ioutil.WriteFile(resultsLogFile, jStr, os.FileMode(0644)); err != nil {
+			glog.Errorf("Error logging results to %s: %v", resultsLogFile, err)
+		}
+	}
+	glog.Infof("AllResults:\n%s", string(jStr))
+}
+
+func TestPerformance(t *testing.T) {
+	apiURL, masterShutdown := util.StartApiserver()
+	defer masterShutdown()
+
+	_, clusterCIDR, _ := net.ParseCIDR("10.96.0.0/11")
+	_, serviceCIDR, _ := net.ParseCIDR("10.94.0.0/18")
+	subnetMaskSize := 24
+
+	allResults := []*Results{}
+
+	// baseline tests for creating 10 notes with 10 qps on kubeAPI for allocator
+
+	t.Run("RangeAllocatorKubeQPS10Nodes10", func(t *testing.T) {
+		config := &Config{
+			NumNodes: 10, CreateQPS: 10, KubeQPS: 10, CloudQPS: 10,
+			AllocatorType: ipam.RangeAllocatorType, Cloud: nil,
+		}
+		if results, err := runTest(t, apiURL, config, clusterCIDR, serviceCIDR, subnetMaskSize); err == nil {
+			allResults = append(allResults, results)
+		}
+	})
+
+	t.Run("IPAMFromClusterAllocatorKubeQPS10Nodes10", func(t *testing.T) {
+		stopCloud, cloud, err := util.NewMockGCECloud(newCloudServer(10))
+		defer stopCloud()
+		if err != nil {
+			t.Fatalf("Error creating mock GCE Cloud: %v", err)
+		}
+		config := &Config{
+			NumNodes: 10, CreateQPS: 10, KubeQPS: 10, CloudQPS: 10,
+			AllocatorType: ipam.IPAMFromClusterAllocatorType, Cloud: cloud,
+		}
+		if results, err := runTest(t, apiURL, config, clusterCIDR, serviceCIDR, subnetMaskSize); err == nil {
+			allResults = append(allResults, results)
+		}
+	})
+
+	t.Run("CloudAllocatorKubeQPS10Nodes10", func(t *testing.T) {
+		cs := newCloudServerWithCIDRAllocator(10, clusterCIDR, subnetMaskSize)
+		stopCloud, cloud, err := util.NewMockGCECloud(cs)
+		defer stopCloud()
+		if err != nil {
+			t.Fatalf("Error creating mock GCE Cloud: %v", err)
+		}
+		config := &Config{
+			NumNodes: 10, CreateQPS: 10, KubeQPS: 10, CloudQPS: 10,
+			AllocatorType: ipam.CloudAllocatorType, Cloud: cloud,
+		}
+		if results, err := runTest(t, apiURL, config, clusterCIDR, serviceCIDR, subnetMaskSize); err == nil {
+			allResults = append(allResults, results)
+		}
+	})
+
+	t.Run("IPAMFromCloudAllocatorKubeQPS10Nodes10", func(t *testing.T) {
+		cs := newCloudServerWithCIDRAllocator(10, clusterCIDR, subnetMaskSize)
+		stopCloud, cloud, err := util.NewMockGCECloud(cs)
+		defer stopCloud()
+		if err != nil {
+			t.Fatalf("Error creating mock GCE Cloud: %v", err)
+		}
+		config := &Config{
+			NumNodes: 10, CreateQPS: 10, KubeQPS: 10, CloudQPS: 10,
+			AllocatorType: ipam.IPAMFromCloudAllocatorType, Cloud: cloud,
+		}
+		if results, err := runTest(t, apiURL, config, clusterCIDR, serviceCIDR, subnetMaskSize); err == nil {
+			allResults = append(allResults, results)
+		}
+	})
+
+	// stress test for creating 100 notes with 10 qps on kubeAPI for allocator
+
+	t.Run("RangeAllocatorKubeQPS10Nodes100", func(t *testing.T) {
+		config := &Config{
+			NumNodes: 100, CreateQPS: 100, KubeQPS: 10, CloudQPS: 10,
```

Comment:
You seem to copy the whole test-case with just these parameters changed. Can you factor this out better? For e.g like here - https://github.com/kubernetes/kubernetes/blob/master/test/integration/scheduler_perf/scheduler_bench_test.go#L57-L60

Reply:
It'll make this file about half it's current size :)